### PR TITLE
Update API Documentation

### PR DESCRIPTION
**Pull request by munishchouhan**
This PR will add the following:
1. Added inspect and build logs API docs
2. updated existing docs

### CODE REVIEW

**Review by pditommaso — APPROVED** [review]

**Comment by jason-seqera** [comment]
@munishchouhan, I've brought over some revision suggestions from the other repo into this PR.

**Comment by munishchouhan** [comment]
@jason-seqera revision looks fine
@pditommaso please merge it

File: docs/api.mdx [hunk 1]
```diff
@@ -6,10 +6,16 @@ This page summarizes the API provided by the Wave container service.
 
 ## POST `/container-token`
 
-This endpoint allows you to submit a request to access a private container repository via Wave, or build a container image on-the-fly with a Dockerfile or Conda recipe file.
+Deprecated endpoint allows you to submit a request to access a private container registry via Wave, or build a container image on-the-fly with a Dockerfile or Conda recipe file.
```
Comment (pditommaso) [inline docs/api.mdx:9]:
```suggestion
Deprecated endpoint allows you to submit a request to access a private container repository via Wave, or build a container image on-the-fly with a Dockerfile or Conda recipe file.
```

Registry is the host, repository is host + project name

**Review by pditommaso — APPROVED** [review]

File: docs/api.mdx (hunk 1 above)
Comment (pditommaso) [reply docs/api.mdx:9]: I've noticed only here

**Review by jason-seqera — APPROVED** [review]

**Comment by jason-seqera** [comment]
@munishchouhan, I think this is still all the correct content. Can you confirm I didn't include/exclude anything by accident? Thanks!

**Pull request merged**